### DISCUSSION TITLE
proposer 2 menus distincts

### DISCUSSION THREAD

**Discussion by laurentsc** [rawsrc/PhpEcho#16] [Q&A]
Bonjour,

je souhaite permettre de switcher entre 2 menus mais tout en ayant un seul fichier **layout.php**. En clair, aujourd'hui celui-ci est codé comme suit : `<!DOCTYPE html>
<html>
<head>
    <base href="<?= URL_LOCALHOST //$this->raw('base_href') ?>">
 
    <link  rel="stylesheet" href="/assets/style.css" media="all" type="text/css">
</head>

<body>
<?= $this->addChildFromCurrent('nav', 'block Nav.php'), $this['body'] ?>
</body>
</html>`
On rajoute donc un menu avec le fichier **Nav.php**. Peut-on modifier le fichier **layout.php** afin de passer un paramètre (debug) au fichier **Nav.php** de telle façon que si **debug** vaut true, alors dans Nav.php on rajoute un onglet de **debug** mais on ne le fait pas si **debug** vaut false ?

**Comment by rawsrc** [comment]
Salut Laurent,

ben tu peux très bien le faire en ajoutant un paramètre au moteur de rendu
:
https://github.com/rawsrc/PhpEcho#parameters

$layout = new PhpEcho(...);
$layout->setParam('debug_mode', true);

Et dans la page de rendu :
if ($this->getParam('debug_mode')) {

} else {

}

Martin

Le dim. 26 sept. 2021 à 20:51, laurentsc ***@***.***> a
écrit :

**Reply by laurentsc** [reply]
Merci pour ta réponse Martin. Mais avec cette solution, je vois une difficulté : j'ai une cinquantaine de fichiers qui instancient la classe PhpEcho. Donc la ligne `$layout->setParam('debug_mode', true);`, il faudra la rajouter 50 fois...C'est pourquoi je me disais que si je pouvais modifier **Layout.php**, la modif ne serait à faire qu'à un seul endroit. Est-ce faisable ?

**Comment by rawsrc** [comment]
euh comment est-ce possible ? tu n'as pas un point d'entrée unique pour toutes les vues ?
Si tu renommes ton fichier layout.php par exemple, tu dois le faire 50 fois ? c'est ça, j'ai bien compris ?

**Comment by laurentsc** [comment]
Tu as bien compris que je code comme un cochon 😢. Il faudrait pas par hasard que je crée une classe **view** qui dispose d'une méthode **render** et que donc pour créer une vue, il suffise d'instancier  **view** et d'appeler la méthode **render** ? (il me semble avoir déjà vu ça).

**Reply by laurentsc** [reply]
Bon, je vais dormir et continuer demain...
